google/cloud/logging_v2/handlers/_monitored_resources.py

```diff
@@ -233,3 +249,55 @@ def detect_resource(project=""):
     else:
         # use generic global resource
         return _create_global_resource(project)
+
+
+@functools.lru_cache(maxsize=None)
+def _get_environmental_labels(resource_type):
+    """Builds a dictionary of labels to be inserted into a LogRecord of the given resource type.
+    This function should only build a dict of items that are consistent across multiple logging statements
+    of the same resource type, such as environment variables.
+
+    Returns:
+        dict:
+            A dict representation of labels and the values of those labels
+    """
+    labels = {}
+
+    def set_item(key, environ_var):
+        val = os.environ.get(environ_var, "")
+        if val:
+            labels[key] = val
+
+    if resource_type == _CLOUD_RUN_JOB_RESOURCE_TYPE:
+        set_item(
+            _CLOUD_RUN_JOBS_EXECUTION_NAME_LABEL,
+            _CLOUD_RUN_EXECUTION_ID,
+        )
+        set_item(_CLOUD_RUN_JOBS_TASK_INDEX_LABEL, _CLOUD_RUN_TASK_INDEX)
+        set_item(_CLOUD_RUN_JOBS_TASK_ATTEMPT_LABEL, _CLOUD_RUN_TASK_ATTEMPT)
```

Comment:
nit: this works, but I find the indirection a bit harder to read than it probably needs to be. Maybe something like:

```
var_map =  ((_CLOUD_RUN_EXECUTION_ID, _CLOUD_RUN_JOBS_EXECUTION_NAME_LABEL), (_CLOUD_RUN_TASK_INDEX, _CLOUD_RUN_JOBS_TASK_INDEX_LABEL), (_CLOUD_RUN_TASK_ATTEMPT, _CLOUD_RUN_JOBS_TASK_ATTEMPT_LABEL))

for envvar, label_name in var_map:
  value = os.environ.get(envvar, None):
  if value:
    labels[label_name] = value
  
```